### PR TITLE
Fix tests so that they also work locally

### PR DESCRIPTION
**Pull request by FranklinYu**
The test works in GitHub Action environment, but it doesn’t work on all environments. This PR is trying to make the test more robust, and allow it to run on more environments (such as the one I develop on).

### CODE REVIEW

File: test/unit/chruby_reset_test.sh [hunk 1]
```diff
@@ -15,10 +15,13 @@ function test_chruby_reset_hash_table()
 		assertEquals "did not clear the path table" \
 			     "" "$(hash)"
 	elif [[ -n "$BASH_VERSION" ]]; then
+		export LANG=en_US.UTF-8
```
Comment (FranklinYu) [inline test/unit/chruby_reset_test.sh:18]: Note: tested on my machine, setting system language to Chinese. Without this change, the error message contains Chinese.

**Comment by FranklinYu** [comment]
Quote from Bash manual supporting last commit:

> If a numeric argument is given to `return`, that is the function’s return status; otherwise the function’s return status is the exit status of the last command executed before the `return`.

Example:

```bash
func() { return; }

false
func || echo 'Error!'
```

File: test/unit/chruby_reset_test.sh (hunk 1 above)
Comment (postmodern) [inline test/unit/chruby_reset_test.sh:18]: We should probably do this for `zsh` as well, yes?

File: test/unit/chruby_reset_test.sh [hunk 2]
```diff
@@ -15,10 +15,13 @@ function test_chruby_reset_hash_table()
...
 		chruby_reset
 
 		assertEquals "did not clear the path table" \
 			     "hash: hash table empty" "$(hash)"
+
+		unset LANG
```
Comment (postmodern) [inline test/unit/chruby_reset_test.sh:24]: `LANG` is usually already set, so many store the original `LANG` value and reset it?

**Review by postmodern — COMMENTED** [review]
Minor question regarding `bash` vs `zsh` with `LANG`.

File: test/unit/chruby_reset_test.sh (hunk 1 above)
Comment (FranklinYu) [reply test/unit/chruby_reset_test.sh:18]: The Zsh test above only compares it with empty string, which doesn’t sound like language-specific.

File: test/unit/chruby_reset_test.sh (hunk 2 above)
Comment (FranklinYu) [reply test/unit/chruby_reset_test.sh]: Done.

**Comment by postmodern** [comment]
Merged!

**Pull request merged**